cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -1345,22 +1250,50 @@ pProposalFiles sbe balExUnits =
 
 pProposalFile
   :: BalanceTxExecUnits
-  -> Parser (ProposalFile In, Maybe (ScriptWitnessFiles WitCtxStake))
+  -> Parser (ProposalFile In, Maybe CliProposalScriptRequirements)
 pProposalFile balExUnits =
   (,)
     <$> pFileInDirection "proposal-file" "Filepath of the proposal."
     <*> optional (pProposingScriptOrReferenceScriptWitness balExUnits)
  where
   pProposingScriptOrReferenceScriptWitness
-    :: BalanceTxExecUnits -> Parser (ScriptWitnessFiles WitCtxStake)
+    :: BalanceTxExecUnits -> Parser CliProposalScriptRequirements
   pProposingScriptOrReferenceScriptWitness bExUnits =
-    pScriptWitnessFiles
-      WitCtxStake
+    pProposalScriptWitness
       bExUnits
       "proposal"
       Nothing
       "a proposal"
-      <|> pPlutusStakeReferenceScriptWitnessFilesVotingProposing "proposal-" balExUnits
+      <|> pProposalReferencePlutusScriptWitness "proposal-" balExUnits
+
+pProposalScriptWitness
+  :: BalanceTxExecUnits -> String -> Maybe String -> String -> Parser CliProposalScriptRequirements
+pProposalScriptWitness bExecUnits scriptFlagPrefix scriptFlagPrefixDeprecated help =
+  Proposing.createSimpleOrPlutusScriptFromCliArgs
+    <$> pScriptFor
+      (scriptFlagPrefix ++ "-script-file")
+      ((++ "-script-file") <$> scriptFlagPrefixDeprecated)
+      ("The file containing the script to witness " ++ help)
+    <*> optional
+      ( (,)
+          <$> pScriptRedeemerOrFile scriptFlagPrefix
+          <*> ( case bExecUnits of
+                  AutoBalance -> pure (ExecutionUnits 0 0)
+                  ManualBalance -> pExecutionUnits scriptFlagPrefix
+              )
+      )
+
+pProposalReferencePlutusScriptWitness
+  :: String -> BalanceTxExecUnits -> Parser CliProposalScriptRequirements
+pProposalReferencePlutusScriptWitness prefix autoBalanceExecUnits =
+  Proposing.createPlutusReferenceScriptFromCliArgs
+    <$> pReferenceTxIn prefix "plutus"
+    <*> plutusP prefix PlutusScriptV3 "v3"
+    <*> pScriptRedeemerOrFile (prefix ++ "reference-tx-in")
+    <*> ( case autoBalanceExecUnits of
+            AutoBalance -> pure (ExecutionUnits 0 0)
```

Comment:
It would be nice to have this execution units default value in one place, instead of being copied multiple times.

Reply:
Yep we will do an audit of all the cli parsers. It's still convoluted.